controller/api/public/prometheus.go

```diff
@@ -64,35 +64,53 @@ func (s *grpcServer) queryProm(ctx context.Context, query string) (model.Vector,
 }
 
 // add filtering by resource type
-// note that metricToKey assumes the label ordering (namespace, name)
 func promGroupByLabelNames(resource *pb.Resource) model.LabelNames {
 	names := model.LabelNames{namespaceLabel}
+	if resource.Type == k8s.Namespace {
+		return names
+	}
 
-	if resource.Type != k8s.Namespace {
-		names = append(names, promResourceType(resource))
+	if resource.Type == k8s.All {
+		for _, kind := range k8s.StatAllWorkloadKinds {
+			names = append(names, promResourceType(kind))
+		}
+		return names
 	}
-	return names
+
+	return append(names, promResourceType(resource.Type))
 }
 
 // add filtering by resource type
-// note that metricToKey assumes the label ordering (namespace, name)
 func promDstGroupByLabelNames(resource *pb.Resource) model.LabelNames {
 	names := model.LabelNames{dstNamespaceLabel}
+	if resource.Type == k8s.Namespace {
+		return names
+	}
 
 	if isNonK8sResourceQuery(resource.GetType()) {
-		names = append(names, promResourceType(resource))
-	} else if resource.Type != k8s.Namespace {
-		names = append(names, "dst_"+promResourceType(resource))
+		return append(names, promResourceType(resource.Type))
 	}
-	return names
+
+	if resource.Type == k8s.All {
+		for _, kind := range k8s.StatAllWorkloadKinds {
+			if isNonK8sResourceQuery(kind) {
```

Comment:
If kind is authority, don't add the `dst_` prefix to the label.

Reply:
But `k8s.StatAllWorkloadKinds` doesn't contain `Authority`, so this will always be false, won't it?

Reply:
Yes, you are right. The `Authority` case is already handled in linnke 90 above, so this is not needed. Thanks.